aptos-move/framework/aptos-framework/sources/native_bridge.move

```diff
@@ -0,0 +1,794 @@
+            amount,
+            nonce
+        );
+
+        let bridge_events = borrow_global<BridgeEvents>(signer::address_of(aptos_framework));
+        let complete_events = event::emitted_events_by_handle(&bridge_events.bridge_transfer_completed_events);
+
+        // Assert that the event was emitted
+        let expected_event = BridgeTransferCompletedEvent {
+            bridge_transfer_id,
+            initiator,
+            recipient,
+            amount,
+            nonce,
+        };
+
+        assert!(std::vector::contains(&complete_events, &expected_event), 0);
+    }
+
+    #[test(aptos_framework = @aptos_framework, sender = @0xdaff)]
+    // #[expected_failure(abort_code = 0x1, location = 0x1::native_bridge_configuration)] // EINVALID_BRIDGE_relayer
+    fun test_complete_bridge_transfer_by_sender(
+        sender: &signer,
+        aptos_framework: &signer
+    ) acquires BridgeEvents, Nonce {
+        let sender_address = signer::address_of(sender);
+        // Create an account for our recipient
+        native_bridge_core::initialize_for_test(aptos_framework);
+        initialize(aptos_framework);
+        aptos_account::create_account(sender_address);
+
+        let recipient = valid_eip55();
+        let amount = 1000;
+        let account_balance = amount + 1;
+
+        // Mint some coins
+        native_bridge_core::mint(sender_address, account_balance);
+
+        assert!(coin::balance<AptosCoin>(sender_address) == account_balance, 0);
+
+        initiate_bridge_transfer(
+            sender,
+            recipient,
+            amount
+        );
+
+        let bridge_events = borrow_global<BridgeEvents>(@aptos_framework);
+        let bridge_transfer_initiated_events = event::emitted_events_by_handle(
+            &bridge_events.bridge_transfer_initiated_events
+        );   
+        let bridge_transfer_initiated_event = vector::borrow(&bridge_transfer_initiated_events, 0);
+
+        let bridge_transfer_id = bridge_transfer_initiated_event.bridge_transfer_id;
+        
+    }
+
+    #[test(aptos_framework = @aptos_framework, sender = @0xdaff)]
+    #[expected_failure(abort_code = EINVALID_BRIDGE_TRANSFER_ID, location = Self)] // ENOT_FOUND
+    fun test_complete_bridge_with_erroneous_bridge_id_by_relayer(
+        sender: &signer,
+        aptos_framework: &signer
+    ) acquires BridgeEvents {
+        let sender_address = signer::address_of(sender);
+        // Create an account for our recipient
+        native_bridge_core::initialize_for_test(aptos_framework);
+        aptos_account::create_account(sender_address);
+
+        let bridge_transfer_id = b"guessing the id";
+
+        // As relayer I send a complete request and it should fail
+        complete_bridge_transfer(
+            aptos_framework,
+            bridge_transfer_id,
+            valid_eip55(),
+            sender_address,
+            1000,
+            1
+        );
+    }
+}
+
+module aptos_framework::native_bridge_store {
```

Comment:
I don't fully understand the logic in the brdge_store and why we need a bridge store at all.
Could you explain why we need that?

Reply:
In `native_bridge_store` the structs (eg `SmartTableWrapper` and `OutboundTransfer`) and functions (eg `create_details`, `add`, etc) are related to storage and processing transfer details, as compared to the `native_bridge` module which has entry functions as an interface for the public (`intiate_bridge_transfer`) and the relayer (`complete_bridge_transfer`).

It does seem to me that the logic in `native_bridge_core` can be consolidated into `native_bridge_store` to simplify the module structure. I'll look into that.